assets/js/app/DataLayer.js

```diff
@@ -98,30 +88,89 @@ LocusZoom.DataLayer.prototype.reMap = function(){
     var promise = this.parent.parent.lzd.getData(this.parent.parent.state, this.fields); //,"ld:best"
     promise.then(function(new_data){
         this.data = new_data.body;
-        this.postget();
     }.bind(this));
     return promise;
 };
 
+/****************
+  Color Functions
+  Singleton for accessing/storing functions to apply different color schemes to data sets
+*/
+
+LocusZoom.DataLayer.ColorFunctions = (function() {
```

Comment:
Agreed - because this branch was already pretty extensive in scope I didn't begin diving into abstracting data layers here. However on the [`abstract_data_layers`](https://github.com/statgen/locuszoom/compare/new_layout_syntax...abstract_data_layers?expand=1#diff-4d12c420a365f5ace0fca5d054d05d6fR276) branch (which branches off of `new_layout_syntax`) this is already implemented.